docs/getting_started.md

```diff
@@ -0,0 +1,325 @@
+```
+
+The `ecs` in the endpoint `api/schema/ecs` is an arbitrary name, it's how
+the loaded schema is addressed by the server.
+
+## Define the data model
+
+In the data model you describe the data that shall be generated.  It can
+be as simple as a list of fields that need to be present or more complex
+for defining also the relations among them.
+
+How to write a data model is separate subject (see [Data model](data_model.md)),
+here we focus on how to configure one on the server. You use the `api/source` endpoint.
+
+```shell
+$ curl -s -XPUT -H "Content-Type: application/yaml" "$GENEVE/api/source/mydata" --data-binary @- <<EOF
+schema: ecs
+queries:
+  - 'network where cidrMatch(destination.ip, "10.0.0.0/8", "192.168.0.0/16")'
+EOF
+```
+
+Note the reference to the previously loaded schema `ecs` and name of this
+newly defined source, `mydata`. Also, `queries` is a list. You can add as
+many queries you need, at each iteration Geneve will select one randomly.
+
+You can generate some data right on terminal for early inspection
+
+```shell
+$ curl -s "$GENEVE/api/source/mydata/_generate?count=1" | jq
+[
+  {
+    "@timestamp": "2023-01-31T18:19:20.197+01:00",
+    "destination": {
+      "ip": "192.168.130.52"
+    },
+    "event": {
+      "category": [
+        "network"
+      ]
+    }
+  }
+]
+```
+
+## Set the destination
+
+Once you're happy with the data model it's time to configure where data
+shall be sent to. Endpoint `api/sink` serves the purpose.
+
+The command is rather unsofisticated:
+
+```shell
+curl -s -XPUT -H "Content-Type: application/yaml" "$GENEVE/api/sink/mydest" --data-binary @- <<EOF
+url: $TARGET_ES/myindex/_doc
+EOF
+```
+
+The generated documents are `POST`ed to the configured url one by one. The
+name of this sink is `mydest`, the destination index is `myindex`.
+
+## Configure the flow
+
+Flow configuration is also quite basic, you just need a source and a sink.
+They need to be already defined in the server.
+
+Use `count` to specify how many documents should be generated and sent to
+the stack. This flow is named `myflow`.
+
+```shell
+$ curl -s -XPUT -H "Content-Type: application/yaml" "$GENEVE/api/flow/myflow" --data-binary @- <<EOF
+source:
+  name: mydata
+sink:
+  name: mydest
+count: 1000
+EOF
+```
+
+All is left to do is to initiate the generation with
+
+```shell
+$ curl -s -XPOST "$GENEVE/api/flow/myflow/_start"
+```
+
+You can also check the progress with 
+
+```shell
+$ curl -s "$GENEVE/api/flow/myflow"
+params:
+    source:
+        name: mydata
+    sink:
+        name: mydest
+    count: 1000
+state:
+    alive: true
+    documents: 250
+    documents_per_second: 350
+```
+
+Or stop it with
+
+```shell
+$ curl -s -XPOST "$GENEVE/api/flow/myflow/_stop"
+```
+
+## Extra steps
+
+Geneve assumes the target stack and index to be ready for documents
+acceptance, it seems pointless and expensive to duplicate the stack and
+indices configuration functionality.
```

Comment:
While this may be the case for established clusters, developer clusters are often freshly stood up and so any assistance in helping get the right mappings configured will go a long way in easing the friction for the developer experience. 

For example, if I just need to trigger this one specific Detection Rule (whether testing a new feature, or debugging an issue), I'm going to need to start geneve, configure a data model, load the rules, grab the query from the rule, generate data, determine what index patterns are configured for the rule, generate the mapping, push the mapping, and finally push the data. 

This is quite a bit of overhead for just triggering a rule, albeit way easier than having to setup an endpoint and trigger manually 😅 (though that's getting even easier now with tools like the agent CLI added in https://github.com/elastic/kibana/pull/149903). 

Ideally though, it would be nice to be able to just specify a Detection Rule name/id, maybe some run configuration (how many times to replay, interval, replacement variables, etc), and then have the system do all of the above. Perhaps that's the job of another intermediary tool as to not pollute geneve with business logic outside of event generation (not sure of the boundaries here), but maybe just having some default data models and helper scripts could get things a little more automated similar to how [Rally Tracks](https://esrally.readthedocs.io/en/stable/adding_tracks.html) works as I mentioned above.

What do you think about the above? Is this a reasonable way to ease this workflow?

Reply:
Good food for thoughts, I already got very positive feedback for such things working out-of-the-box.

I would stay away from the complexity of Rally tracks, where are so many moving parts, and embrace the "progressive complexity" path where you first get something working and then you learn how to tweak it along your needs.

Although triggering alerts is the reason Geneve was conceived, it's not its only possible application any more. I think that a specific tool based on Geneve and maybe even distributed as part of Geneve (why not?) could serve such users best. You know, small tools that do their job well.

Reply:
Created https://github.com/elastic/geneve/issues/130, feel free to add your desiderata there.